include/xeus-python/xinterpreter.hpp

```diff
@@ -73,6 +73,7 @@ namespace xpyt
         nl::json internal_request_impl(const nl::json& content) override;
 
         void redirect_output();
+        void instanciate_ipython_shell();
```

Comment:
Since you redefine it in the `wasm_interpreter` class, this method should be virtual (and use override in the inheriting class to avoid signature silent errors). Besides, if the implementations of this method in inheriting classes never calls the implementation of the mother class, this method should probably be private.